src/calendar.rs

```diff
@@ -135,7 +154,13 @@ mod tests {
     #[allow(clippy::unreadable_literal)] // TODO: Make this clear.
     #[test]
     fn test_time_from_ymdhms_utc() {
-        use super::{time_from_ymdhms_utc, Time};
+        use super::{time_from_ymdhms_utc, Error, Time, UNIX_EPOCH_YEAR};
+
+        // Before 1970
+        assert_eq!(
+            Err(Error::BadDerTime),
+            time_from_ymdhms_utc(UNIX_EPOCH_YEAR - 1, 1, 1, 0, 0, 0)
+        );
```

Comment:
This seems OK to me, but I'd also like to see test cases for the very edge cases: the moment before the first of 1970 and exactly 1970.

Reply:
I changed the tests to cover:

 * 1969-12-31 00:00:00
 * 1969-12-31 23:59:59
 * 1970-01-01 00:00:00
 * 1970-01-01 00:00:01
 * 1971-01-01 00:00:00

This would catch off by one errors in date calculations.